src/abstractions/serialization/KiotaJsonSerializer.Serialization.cs

```diff
@@ -20,9 +20,21 @@ public static partial class KiotaJsonSerializer
     /// </summary>
     /// <typeparam name="T">Type of the object to serialize</typeparam>
     /// <param name="value">The object to serialize.</param>
+    /// <param name="serializeOnlyChangedValues">By default you'll only get the changed properties</param>
     /// <returns>The serialized representation as a stream.</returns>
-    public static Stream SerializeAsStream<T>(T value) where T : IParsable
-    => KiotaSerializer.SerializeAsStream(_jsonContentType, value);
+    public static Stream SerializeAsStream<T>(T value, bool serializeOnlyChangedValues = true) where T : IParsable
+    => KiotaSerializer.SerializeAsStream(_jsonContentType, value, serializeOnlyChangedValues);
+
+    /// <summary>
+    /// Serializes the specified object as a string based JSON stream.
+    /// </summary>
+    /// <typeparam name="T">The type of the value to serialize.</typeparam>
+    /// <param name="value">The object to serialize.</param>
+    /// <param name="serializeOnlyChangedValues">By default you'll only get the changed properties</param>
+    /// <returns>A <see cref="Stream"/> containing the serialized JSON data.</returns>
+
+    public static Stream SerializeAsJsonStream<T>(this T value, bool serializeOnlyChangedValues = true) where T : IParsable
```

Comment:
I originally did not catch the fact this was an extension method. 
Should we roll this change back? Because it's going to provide a different experience between the different data types. Also with other languages.

Reply:
Hmm... I explicitly choose it to be an extension method, this will help discovery.

That other languages don't support extension methods, resulting in not using extension methods every where, seems a political decision (which I would strongly disagree with).
At best this difference would convince developers to start thinking about switching, and at worst developers could be disappointed in the languages that does not support extension methods.

Reply:
I don't think we want to be in the business of convincing people to switch to dotnet.
Does the IEnumerable<IParsable> extension method work for List<Parsable> or Parsable[] for example? (I have had surprises with generics and covariance in the past)
Should they be under a different namespace to avoid potential collision/confusion? (.Extensions)

Reply:
It will work on list not on `IParsable[]` because it does not see an array as a generic Ienumerable (last time I checked)

Reply:
Thank you for the additional information. I think this is what is bothering me the most: the differences in experiences between the different types:
- Parsable -> stream: extension works
- IEnumerable of Parsable -> stream: extension works
- Parsable[] -> stream: extension doesn't work
- Parsable -> string: no extension
- IEnumerable of Parsable -> string: no extension
- Parsable[] -> string: no extension

Also, why would we have extension methods only for Json?

Now, if we wanted to cover everything, we'd need lots of additional methods. At which point I still believe they should be in their own file/namespace.
The alternative being: switching all the existing methods to ALSO be extension methods, they should still work as a static method.
Both approaches won't solve the array aspect, but that could be solved by adding additional extension methods to do the mapping.

Thoughts?

Reply:
In the latest update I put them in their respective projects.

If you're like me and did remove some of the serializers, you won't need those packages included as well. If you would then use the `KiotaJsonSerializer`, it would throw an exception. In the new case you'll need the actual using to `Microsoft.Kiota.Serialization.Json` and each IParsable will have `SerializeAsStream` and `SerializeAsJsonStringAsync` (and arrays and ienumerables)

Reply:
The addition of the methods (at least for Json) introduces some duplication with the existing helpers from Abstractions (I think we'd at least need to create an issue to come clean this up later on the next major version if we go this route). 
I do like the suggestion by @baywet to convert the helpers to extension methods (we can still have consistency on other languages and call them the same way as now).

> If you're like me and did remove some of the serializers, you won't need those packages included as well. If you would then use the KiotaJsonSerializer it would throw an exception.

The problem here would be registration of the custom serializer and not where the code would reside. If the custom serilizer is registered, it should not matter where we are at.

Reply:
We could create them as extension methods. But those need the object you want to convert to be the first parameter. That would be a source breaking change.

Reply:
in the interest of getting this pull request merged can we please:
- create another issue to bring the extension methods
- remove any extension method from this PR

Reply:
@baywet done